### ISSUE TITLE
Persist changes to an input field in the browser to the file system

### ISSUE THREAD

**Issue by arekkas** [gatsbyjs/gatsby#2033]
Suppose we have an input field in one of our components `<input onChange={doSomething()} value={data.jsonFile/*.nodes.foo.bar*/}>`. Now, I want to be able to update this input field, and write the contents to a json file. The contents of this json file will be queried using GraphQL.

Is it possible to reload the page only when the JSON file is changed from outside of gatsby (e.g. with a text editor) as opposed to not reloading when the user updates the input field?

**Comment by MarcCoet** [comment]
Not sure I understand your question.
Keep in mind that Gatsby is a static site generator so if you want to modify your data (a JSON file on your file system) you will generally need to rebuild your website.
Am I missing the point?

**Comment by arekkas** [comment]
Yeah, let me rephrase the question - is it possible to update the state of e.g. a node from within a component? So basically calling a redux action creator using `react-redux` to update a node / page / layout / whatever.

**Comment by MarcCoet** [comment]
Hmmm... Don't think so. As far as I know [action creators](https://www.gatsbyjs.org/docs/bound-action-creators/) can only manipulate state during build and are implemented in your `gatsby-node.js`, in plugins code, etc.

**Comment by arekkas** [comment]
Ok thanks for the clarification. I guess it should be possible to wrap the layout component in a ReduxProvider component with my own store and manage the state on my own - and on save writing those changes to the filesystem (or a database or whatever).

**Comment by KyleAMathews** [comment]
Yeah it's just react so use whatever you normally do in the client, setState, redux, mobx, graphql, etc.

You just need to make data available to Gatsby if you want it included at build time.

**Comment by MarcCoet** [comment]
And there is a basic redux example [here](https://github.com/gatsbyjs/gatsby/tree/master/examples/redux).

**Comment by arekkas** [comment]
Thank you!
